### PR TITLE
Include mention of the onError Prop for next/script

### PR DESCRIPTION
**Pull request by mix3d**
None of the documentation includes this, despite being written into the code.
I might also suggest mentioning the function signature change somewhere between v11 and v12 that now includes the error object. (Was confused when my TS definitions were complaining about my onError handler, reading the repo's code while using the v11.1.x version in my project)

## Documentation / Examples

- [X] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/basic-features/script.md [hunk 1]
```diff
@@ -174,6 +174,26 @@ export default function Home() {
 }
 ```
 
+Sometimes it is helpful to catch when a script fails to load. These events can be caught with the `onError` property:
```
Comment (styfle) [inline docs/basic-features/script.md:177]:
```suggestion
Sometimes it is helpful to catch when a script fails to load. These errors can be handled with the `onError` property:
```

File: docs/basic-features/script.md [hunk 2]
```diff
@@ -174,6 +174,26 @@ export default function Home() {
...
+
+```jsx
+import Script from 'next/script'
+
+export default function Home() {
+  return (
+    <>
+      <Script
+        id="will-fail"
+        src="https://non-existant-file.com/script.js"
```
Comment (styfle) [inline docs/basic-features/script.md:187]:
```suggestion
        src="https://example.com/non-existant-script.js"
```

File: docs/basic-features/script.md [hunk 3]
```diff
@@ -174,6 +174,26 @@ export default function Home() {
 }
 ```
```
Comment (styfle) [inline docs/basic-features/script.md:176]: @molebox This should probably be its own section, right?

File: docs/basic-features/script.md (hunk 1 above)
Comment (mix3d) [reply docs/basic-features/script.md:177]: I'm not sure I agree with your suggested changes. Why do you think "error" is better than "event"?

Comment (styfle) [reply docs/basic-features/script.md:177]: The returned parameter is an error object

Comment (mix3d) [reply docs/basic-features/script.md:177]: fair.

File: docs/basic-features/script.md (hunk 3 above)
Comment (mix3d) [reply docs/basic-features/script.md:176]: I put it with the Script Loading stuff. Happy to move wherever.

**Review by styfle — DISMISSED** [review]

**Review by ijjk — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 16.5s | 16.7s | ⚠️ +180ms |
| buildDurationCached | 3s | 3.2s | ⚠️ +184ms |
| nodeModulesSize | 350 MB | 350 MB | -5 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.643 | 2.884 | ⚠️ +0.24 |
| / avg req/sec | 946.02 | 866.9 | ⚠️ -79.12 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.238 | 1.256 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2019.58 | 1989.8 | ⚠️ -29.78 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.2 kB | 30.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74 kB | 74 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| buildDuration | 17.7s | 17.9s | ⚠️ +181ms |
| buildDurationCached | 3.2s | 3.1s | -46ms |
| nodeModulesSize | 350 MB | 350 MB | -5 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.705 | 2.711 | ⚠️ +0.01 |
| / avg req/sec | 924.15 | 922.2 | ⚠️ -1.95 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.275 | 1.308 | ⚠️ +0.03 |
| /error-in-render avg req/sec | 1961.46 | 1910.91 | ⚠️ -50.55 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.3 kB | 74.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | mix3d/next.js patch-1 | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: 15a7b0d66808ee66e179adb1651a9cab23f061bb

**Pull request merged**